javascript/implementation/Database.ts

```diff
@@ -81,10 +73,34 @@ export class Database {
                 break;
             }
         }
-        if (!this.myChain) {
-            await this.startChain(identity);
+        if (reused) {
+            ensure(reused.medallion > 0);
+            this.myChain = reused;
+        } else {
+            const medallion = makeMedallion();
+            const chainStart = generateTimestamp();
+            const bundler = new Bundler(this.identity, medallion);
+            const bundleInfo = bundler.seal({
+                medallion, timestamp: chainStart, chainStart
+            });
+            ensure(bundleInfo.comment == this.identity);
+            const commitBytes = bundler.bytes;
+            await this.store.addBundle(commitBytes, true);
+            this.myChain = (await this.store.getClaimedChains()).get(medallion);
+            this.iHave.markAsHaving(bundleInfo);
+            // If there is already a connection before we claim a chain, ensure the
+            // peers get this commit as well so future commits will be valid extensions.
+            for (const [peerId, peer] of this.peers) {
+                peer._sendIfNeeded(commitBytes, bundleInfo);
```

Comment:
This was the last piece to get this to actually work. Since there may already be connections when we start a new chain, the peers need to know too. This was causing a ton of invalid extension errors before I added this.